cocotb/handle.py

```diff
@@ -415,12 +416,13 @@ def __le__(self, value):
         return AssignmentResult(self, value)
 
     def __eq__(self, other):
-        if isinstance(other, SimHandleBase):
-            if self._handle == other._handle:
-                return 0
-            return 1
+        """Equality comparator for non hierarchy handles
 
-        # Use the comparison method of the other object against our value
+        If other is not SimHandleBase instance, use comparison method of the
+        other object against our value
+        """
+        if isinstance(other, SimHandleBase):
+            return SimHandleBase.__eq__(self, other)
```

Comment:
This should be `return NotImplemented`, which will automatically call the mro-respecting `__eq__` of `other`

Reply:
Wait, nevermind, I read that backwards

Reply:
Possibly clearer as `return super(NonHierarchyObject, self).__eq__(other)`

Reply:
I'm not sure - I like the `super` syntax and would typically use it, but this is checking if it's a child of the parent-est class, so makes sense to me to use the same handle as in the conditional statement when calling `__eq__`